File: Source/STSL.SmartLocker.Utils.Kiosk.Client/src/models/KioskAccessCode.ts
import type { EntityId, IUsesGuidId } from "./Common";

export interface IKioskAccessCode extends IUsesGuidId {
    accessCode: string
    hasBeenUsed: boolean
    expiryDate: Date
    cardHolderId: EntityId
}


export interface IAccessCode {
    code: string
}

File: Source/STSL.SmartLocker.Utils.Domain/Kiosk/KioskAccessCode.cs
namespace STSL.SmartLocker.Utils.Domain.Kiosk;

public class KioskAccessCode : EntityBaseWithTenant, IUsesGuidId
{
    public Guid Id { get; set; }
    public required string AccessCode { get; set; }
    public bool HasBeenUsed { get; set; } = false;
    public DateTime ExpiryDate { get; set; }
    public Guid CardHolderId { get; set; }
    public CardHolder? CardHolder { get; set; }
}